hierarchicalforecast/core.py

```diff
@@ -92,6 +92,8 @@ def reconcile(self,
         uids = Y_hat_df.index.unique()
         # check if Y_hat_df has the same uids as S
         if len(S.index.difference(uids)) > 0 or len(Y_hat_df.index.difference(S.index.unique())) > 0:
+            print(len(S.index.difference(uids)))
+            print(len(Y_hat_df.index.difference(S.index.unique())))
```

Comment:
Please add a description of what is being printed. Those prints should be included in the `Exception` statement. For multiple lines, the following syntax can be used,

```
raise Exception(
        'line 1 '
        'line 2 '
        'line 3 ' 
)
````

Reply:
Also, maybe we can avoid doing the calculation of the differences twice. Something like,

```
S_diff_uids  = S.index.difference(uids)
Y_hat_df_diff_uids = Y_hat_df.index.difference(S.index.unique())

if len(S_diff_uids) > 0 or len(Y_hat_df_diff_uids) > 0:
    raise Exception(print lens of S and Y_hat diffs)
```

Reply:
Thanks I added the recommendations and complimented with extra insample `Y_df` protection.